tests/ci/ci_utils.py

```diff
@@ -215,6 +215,20 @@ def is_latest_release_branch(branch):
         )
         return latest_branch == branch
 
+    @staticmethod
+    def is_latest_branch_per_first_version_segment(branch):
+        release_branches = Shell.get_output(
+            'gh pr list --label release --repo ClickHouse/ClickHouse --search "sort:created" --json headRefName'
```

Comment:
It belongs to `version_helper`. There are [get_tagged_versions](https://github.com/ClickHouse/ClickHouse/blob/0ff7c280f9e032732b27f42bd5f5b5677e9fd723/tests/ci/version_helper.py#L353) and [get_supported_versions](https://github.com/ClickHouse/ClickHouse/blob/0ff7c280f9e032732b27f42bd5f5b5677e9fd723/tests/ci/version_helper.py#L364). It feels wrong to request the gh for this case.

Reply:
it's an alternative which does not require full git history

Reply:
but the whole workflow requires the full history. How do we set the tag to a commit otherwise?